libraries/libstratosphere/source/fs/fs_code.cpp

```diff
@@ -20,6 +20,55 @@ namespace ams::fs {
 
     namespace {
 
+        constinit os::SdkMutex g_mount_stratosphere_romfs_lock;
+        constinit bool g_mounted_stratosphere_romfs = false;
+
+        constinit TYPED_STORAGE(FileHandleStorage) g_stratosphere_romfs_storage = {};
+        constinit TYPED_STORAGE(RomFsFileSystem) g_stratosphere_romfs_fs = {};
+
+        Result EnsureStratosphereRomfsMounted() {
+            std::scoped_lock lk(g_mount_stratosphere_romfs_lock);
+
+            if (AMS_UNLIKELY(!g_mounted_stratosphere_romfs)) {
+                /* Mount the SD card. */
+                R_TRY(fs::MountSdCard("#strat-romfs-sd"));
+                auto sd_guard = SCOPE_GUARD { fs::Unmount("#strat-romfs-sd"); };
+
+                /* Open sd:/atmosphere/stratosphere.romfs. */
+                fs::FileHandle stratosphere_romfs_file;
+                R_TRY(fs::OpenFile(std::addressof(stratosphere_romfs_file), "#strat-romfs-sd:/atmosphere/stratosphere.romfs", fs::OpenMode_Read));
+
+                /* Setup the storage. */
+                /* NOTE: This owns the file, and so on failure it will be closed appropriately. */
+                std::construct_at(GetPointer(g_stratosphere_romfs_storage), stratosphere_romfs_file, true);
+                auto storage_guard = SCOPE_GUARD { std::destroy_at(GetPointer(g_stratosphere_romfs_storage)); };
```

Comment:
Seems like there should be a pattern for constructing into a storage and creating a guard for it all in one function.